### PR TITLE
Source Facebook Marketing: fix permission error

### PR DESCRIPTION
**Pull request by roman-yermilov-gl**
## What
*https://github.com/airbytehq/oncall/issues/3167*

## How
*AdAccount stream: catch an error and re-run the request without `owner` field*

## Why?
*The error says: **"Requires business_management permission to manage the object"** when account has all the required permissions:*
```
[
    'ads_management',
    'ads_read',
    'business_management',
    'public_profile'
]
```

### CODE REVIEW

**Comment by airbyte-oss-build-runner** [comment]
##  source-facebook-marketing test report (commit [`9191cec5d5`](https://github.com/airbytehq/airbyte/pull/31132/commits/9191cec5d5895c3316fdca34a4df980811879a43)) - ❌

⏲️  Total pipeline duration: 135mn30s 

| Step                                                                      | Result   |
|:--------------------------------------------------------------------------|:---------|
| Connector package install                                                 | ✅        |
| Build source-facebook-marketing docker image for platform(s) linux/x86_64 | ✅        |
| Unit tests                                                                | ✅        |
| Integration tests                                                         | ✅        |
| Acceptance tests                                                          | ❌        |
| Code format checks                                                        | ✅        |
| Validate metadata for source-facebook-marketing                           | ✅        |
| Connector version semver check                                            | ✅        |
| Connector version increment check                                         | ✅        |
| QA checks                                                                 | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/ryermilov_source-facebook-marketing-fix-permission-error/1696615789/9191cec5d5895c3316fdca34a4df980811879a43/source-facebook-marketing/1.1.15/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:9191cec5d5895c3316fdca34a4df980811879a43)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-facebook-marketing test
```

File: airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/streams/streams.py [hunk 1]
```diff
@@ -187,15 +188,25 @@ def fields(self) -> List[str]:
...
-        if "is_prepay_account" in properties and "MANAGE" not in self.get_task_permissions():
+        if "is_prepay_account" in properties and "MANAGE" not in permissions:
             properties.remove("is_prepay_account")
         return properties
 
     def list_objects(self, params: Mapping[str, Any]) -> Iterable:
         """noop in case of AdAccount"""
-        return [FBAdAccount(self._api.account.get_id()).api_get(fields=self.fields)]
+        fields = self.fields
+        try:
```
Comment (davydov-d) [inline airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/streams/streams.py:201]: let's add a unit test for this case please

**Review by davydov-d — CHANGES_REQUESTED** [review]

**Review by davydov-d — APPROVED** [review]

**Comment by airbyte-oss-build-runner** [comment]
##  source-facebook-marketing test report (commit [`ed96a25161`](https://github.com/airbytehq/airbyte/pull/31132/commits/ed96a251617c58f7cd836a32f7a60cbd80a06fef)) - ❌

⏲️  Total pipeline duration: 98mn35s 

| Step                                                                      | Result   |
|:--------------------------------------------------------------------------|:---------|
| Connector package install                                                 | ✅        |
| Build source-facebook-marketing docker image for platform(s) linux/x86_64 | ✅        |
| Unit tests                                                                | ✅        |
| Integration tests                                                         | ✅        |
| Acceptance tests                                                          | ❌        |
| Code format checks                                                        | ✅        |
| Validate metadata for source-facebook-marketing                           | ✅        |
| Connector version semver check                                            | ✅        |
| Connector version increment check                                         | ✅        |
| QA checks                                                                 | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/ryermilov_source-facebook-marketing-fix-permission-error/1696852066/ed96a251617c58f7cd836a32f7a60cbd80a06fef/source-facebook-marketing/1.1.15/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:ed96a251617c58f7cd836a32f7a60cbd80a06fef)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-facebook-marketing test
```

**Comment by airbyte-oss-build-runner** [comment]
##  source-facebook-marketing test report (commit [`d7b35d392d`](https://github.com/airbytehq/airbyte/pull/31132/commits/d7b35d392d02e66dac06c73780498a8acc77225e)) - ❌

⏲️  Total pipeline duration: 97mn07s 

| Step                                                                      | Result   |
|:--------------------------------------------------------------------------|:---------|
| Connector package install                                                 | ✅        |
| Build source-facebook-marketing docker image for platform(s) linux/x86_64 | ✅        |
| Unit tests                                                                | ✅        |
| Integration tests                                                         | ✅        |
| Acceptance tests                                                          | ❌        |
| Code format checks                                                        | ✅        |
| Validate metadata for source-facebook-marketing                           | ✅        |
| Connector version semver check                                            | ✅        |
| Connector version increment check                                         | ✅        |
| QA checks                                                                 | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/ryermilov_source-facebook-marketing-fix-permission-error/1696858573/d7b35d392d02e66dac06c73780498a8acc77225e/source-facebook-marketing/1.1.15/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:d7b35d392d02e66dac06c73780498a8acc77225e)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-facebook-marketing test
```

**Comment by airbyte-oss-build-runner** [comment]
##  source-facebook-marketing test report (commit [`2752850325`](https://github.com/airbytehq/airbyte/pull/31132/commits/275285032567c3706e85bdda8d6f1b6241b4baab)) - ✅

⏲️  Total pipeline duration: 80mn51s 

| Step                                                                      | Result   |
|:--------------------------------------------------------------------------|:---------|
| Connector package install                                                 | ✅        |
| Build source-facebook-marketing docker image for platform(s) linux/x86_64 | ✅        |
| Unit tests                                                                | ✅        |
| Integration tests                                                         | ✅        |
| Acceptance tests                                                          | ✅        |
| Code format checks                                                        | ✅        |
| Validate metadata for source-facebook-marketing                           | ✅        |
| Connector version semver check                                            | ✅        |
| Connector version increment check                                         | ✅        |
| QA checks                                                                 | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/ryermilov_source-facebook-marketing-fix-permission-error/1697014370/275285032567c3706e85bdda8d6f1b6241b4baab/source-facebook-marketing/1.1.15/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:275285032567c3706e85bdda8d6f1b6241b4baab)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-facebook-marketing test
```

**Pull request merged**